Packs/McAfeeNSM/Integrations/McAfeeNSMv2/McAfeeNSMv2.py

```diff
@@ -1831,19 +2124,321 @@ def export_pcap_file_command(client: Client, args: Dict) -> List:
+    """
+    device_id = arg_to_number(args.get('device_id'))
+    if not device_id and device_id != 0:
+        raise DemistoException('Please provide a device_id.')
+
+    response = client.get_device_configuration_request(device_id=device_id)
+    capitalize_response = capitalize_key_first_letter([response])[0]
+    flatterned_response = flatten_and_capitalize(main_dict=capitalize_response,
+                                                 inner_dict_key='PendingChanges')
+
+    readable_output = tableToMarkdown(
+        name='Device Configuration', t=flatterned_response, removeNull=True
+    )
+
+    return CommandResults(
+        readable_output=readable_output,
+        outputs_prefix='NSM.DeviceConfiguration',
+        outputs=flatterned_response,
+        raw_response=response
+    )
+
+
+@polling_function(name='nsm-deploy-device-configuration', interval=INTERVAL, requires_polling_arg=False)
+def deploy_device_configuration_command(args: Dict, client: Client) -> PollResult:
+    """
+    Deploy the configuration of a device.(e.g activate pending changes)
+    Args:
+        args (Dict): - The function arguments.
+        client (Client): - A McAfeeNSM client.
+
+    Returns:
+        A PollResult object with a success or failure message
+    """
+
+    request_id = arg_to_number(args.get('request_id'))
+    device_id = arg_to_number(args.get('device_id'))
+    if not device_id and device_id != 0:
+        raise DemistoException('Please provide a device_id.')
+    if not request_id:       # if this is the first time the function is called
+        is_ssl_push_required = argToBoolean(args.get('push_ssl_key', False))
+        is_gam_update_required = argToBoolean(args.get('push_gam_updates', False))
+        is_sigset_config_push_required = argToBoolean(args.get('push_configuration_signature_set', False))
+        is_botnet_push_required = argToBoolean(args.get('push_botnet', False))
+
+        if not any([is_ssl_push_required, is_gam_update_required, is_sigset_config_push_required, is_botnet_push_required]):
+            raise DemistoException("Please provide at least one argument to deploy")
+
+        requests_id = client.deploy_device_configuration_request(device_id=device_id, isSSLPushRequired=is_ssl_push_required,
+                                                                 isGAMUpdateRequired=is_gam_update_required,
+                                                                 isSigsetConfigPushRequired=is_sigset_config_push_required,
+                                                                 isBotnetPushRequired=is_botnet_push_required).get('RequestId')
+        args["request_id"] = requests_id
+    status = client.check_deploy_device_configuration_request_status(device_id=device_id,
+                                                                     request_id=request_id)
+
+    fail_or_success_list = []
+    build_a_massage = ""
+    for k, v in args.items():
```

Comment:
let's take this whole block out to a helper function, and also add a unit test for it.